### ISSUE TITLE
does anybody know why i have bad prediction results with 91% accuracy?

### ISSUE THREAD

**Issue by zdd0819** [divamgupta/image-segmentation-keras#82]
does anybody know why i have bad prediction results with 91% accuracy? The predict pictures are as following.

![img-2025的副本](https://user-images.githubusercontent.com/39216094/61216634-bd0e0b00-a705-11e9-99f5-dd205282f1e2.jpg)
![img-174](https://user-images.githubusercontent.com/39216094/61216635-bd0e0b00-a705-11e9-8655-f075f710ab3c.jpg)
![img-2025](https://user-images.githubusercontent.com/39216094/61216643-c13a2880-a705-11e9-8c22-be6e0bc781d9.jpg)
![img-174的副本](https://user-images.githubusercontent.com/39216094/61216644-c1d2bf00-a705-11e9-8708-51f4e38d3511.jpg)

**Comment by divamgupta** [comment]
For your case, the model is classifying 91% pixels correctly. For example, if classes such as floor and wall ( which cover majority of the area ) are correctly classified, the accuracy will be still high.

**Comment by zdd0819** [comment]
> For your case, the model is classifying 91% pixels correctly. For example, if classes such as floor and wall ( which cover majority of the area ) are correctly classified, the accuracy will be still high.

So should I contiue training or add more training pictures?

**Comment by divamgupta** [comment]
You can try out diffrent models, you can also try changing the optimiser and other hyper paramaters. IoU is a better metric compared to accuracy.

**Comment by zdd0819** [comment]
> You can try out diffrent models, you can also try changing the optimiser and other hyper paramaters. IoU is a better metric compared to accuracy.

I will try it. Thanks a lot!
By the way, can I find the IoU in this code?

**Comment by dcc222** [comment]
> > You can try out diffrent models, you can also try changing the optimiser and other hyper paramaters. IoU is a better metric compared to accuracy.
> 
> I will try it. Thanks a lot!
> By the way, can I find the IoU in this code?

You can find the IoU and MIou under keras_segmentation/predict.py,which contains def evaluate.

**Comment by zdd0819** [comment]
Thanks!


------------------ Original ------------------

**Comment by dcc222** [comment]
Rencently,I also studied this code.Next,I will  train 2012_VOC datasets by MIoU index.What results would you get from this code .We can enjoy our index of MIoU if you get this from VOC or cityspaces and so on.

**Comment by divamgupta** [comment]
You might not get extremely good results on VOC if you train from scratch, with the current hyperparamaters.  We provide pre-trained models for VOC , Cityscapes which should yield similar IoU stated in the PSPNet paper.
